ddtrace/propagation/http.py

```diff
@@ -1130,24 +1141,29 @@ def my_controller(url, headers):
         :param dict headers: HTTP headers to extract tracing attributes.
         :return: New `Context` with propagated attributes.
         """
-        if not headers:
-            return Context()
+        context = Context()
+        if not headers or config._propagation_behavior_extract == _PROPAGATION_BEHAVIOR_IGNORE:
```

Comment:
Do we need to do this here? 

It will be more performant if we check if `config._propagation_behavior_extract == _PROPAGATION_BEHAVIOR_IGNORE` in `ddtrace.settings.config.py`. This way we can only do it once. Then we disable propagation in one place and log a warning if a customer is setting conflicting settings.